tools/run_tests/jobset.py

```diff
@@ -101,15 +101,22 @@ def message(tag, msg, explanatory_text=None, do_newline=False):
     print '%s: %s' % (tag, msg)
     return
   try:
-    sys.stdout.write('%s%s%s\x1b[%d;%dm%s\x1b[0m: %s%s' % (
-        _BEGINNING_OF_LINE,
-        _CLEAR_LINE,
-        '\n%s' % explanatory_text if explanatory_text is not None else '',
-        _COLORS[_TAG_COLOR[tag]][1],
-        _COLORS[_TAG_COLOR[tag]][0],
-        tag,
-        msg,
-        '\n' if do_newline or explanatory_text is not None else ''))
+    if sys.stdout.isatty():
```

Comment:
Can you merge the non tty path with the windows path? They both do essentially the same thing, but differently